app/models/permissions/technician_permission.rb

```diff
@@ -5,10 +5,19 @@
 #
 module Permissions
   class TechnicianPermission < ScientistPermission
+    # - Technician's cannot edit Admin users or set users to admin type
+    # - Technician's cannot edit the protected flag within Locations or create protected locations
     def initialize(user)
       super
       allow :move_locations, [:create]
-      allow :empty_locations, [:create]
+      allow :location_types, [:create, :update]
+      allow :teams, [:create, :update]
+      allow :users, [:create, :update] do |record|
+        record.user.type != "Administrator" && !record.user.instance_of?(Administrator)
```

Comment:
I'm not quite sure what is going on here?? Can it just be:
`!record.user.administrator?`

Reply:
will have a look, this was my original logic but your suggestion might be equivalent

Reply:
after looking at this it needs to have the logic record.user.instance_of?(Administrator) to catch the case if a technician changes an  admin to have type technician as .administrator? only checks the type which would be technician, but the object class would still be Administrator until the transaction finishes. Hope this makes sense

Reply:
OK. Unfortunately it doesn't makes sense which is slightly concerning. Maybe add a comment as to why it is like that?